x/ibc/07-tendermint/misbehaviour.go

```diff
@@ -59,8 +60,27 @@ func CheckMisbehaviourAndUpdateState(
 // checkMisbehaviour checks if the evidence provided is a valid light client misbehaviour
 func checkMisbehaviour(
 	clientState types.ClientState, consensusState types.ConsensusState, evidence types.Evidence,
-	height uint64, currentTimestamp time.Time,
+	height uint64, currentTimestamp time.Time, consensusParams *abci.ConsensusParams,
 ) error {
+	// calculate the age of the misbehaviour evidence
+	infractionHeight := evidence.GetHeight()
+	infractionTime := evidence.GetTime()
+	ageDuration := currentTimestamp.Sub(infractionTime)
+	ageBlocks := height - uint64(infractionHeight)
+
+	// Reject misbehaviour if the age is too old. Evidence is considered stale
+	// if the difference in time and number of blocks is greater than the allowed
+	// parameters defined.
+	if consensusParams != nil &&
```

Comment:
are `consensusParams` allowed to be nil? I feel like this would be worthy of an error at the very least. 

`consensusParams.Evidence` could probably be nil?, I'm not sure what the implications are of a chain not setting evidence params in relation to light clients

Reply:
the context returns a pointer of the consensus params. They cannot be `nil` (unless set as nil during chain initialization) since the `baseapp.StoreConsensusParams` won't succeed 

> consensusParams.Evidence could probably be nil?, I'm not sure what the implications are of a chain not setting evidence params in relation to light clients

Do you think we should add a check for that during IBC Client InitChain to check that the consensus params and Evidence are set?

```
// StoreConsensusParams sets the consensus parameters to the baseapp's param store.
func (app *BaseApp) StoreConsensusParams(ctx sdk.Context, cp *abci.ConsensusParams) {
	if app.paramStore == nil {
		panic("cannot store consensus params with no params store set")
	}

	if cp == nil {
		return
	}

	app.paramStore.Set(ctx, ParamStoreKeyBlockParams, cp.Block)
	app.paramStore.Set(ctx, ParamStoreKeyEvidenceParams, cp.Evidence)
	app.paramStore.Set(ctx, ParamStoreKeyValidatorParams, cp.Validator)
}
```

Reply:
Hmm, I think so.. Would like @cwgoes and @AdityaSripal input on this

Reply:
The `nil` check is just a safety check as it's theoretically possible not to have consensus params? I'm not sure if Tendermint would allow this though.

Reply:
@marbar3778 @melekes?

Reply:
> The nil check is just a safety check as it's theoretically possible not to have consensus params? I'm not sure if Tendermint would allow this though.

are u updating params here? if so, if some part of ConsensusParams is nil, the previous value is used.

Reply:
> are u updating params here?
no just getting the `MaxAge` values

Reply:
> can they be nil on initialization?

no, Tendermint would use the default values

Reply:
I'll add that as a comment then 👍